internal/pkg/cli/app_init.go

```diff
@@ -0,0 +1,254 @@
+	if err != nil {
+		return fmt.Errorf("relative path of manifest file: %w", err)
+	}
+	opts.manifestPath = relPath
+	log.Infoln()
+	log.Successf("Wrote the manifest for %s app at '%s'\n", color.HighlightUserInput(opts.AppName), color.HighlightResource(opts.manifestPath))
+	log.Infoln("Your manifest contains configurations like your container size and ports.")
+	log.Infoln()
+	return nil
+}
+
+func (opts *AppInitOpts) askAppType() error {
+	t, err := opts.prompt.SelectOne(
+		"Which type of infrastructure pattern best represents your application?",
+		`Your application's architecture. Most applications need additional AWS resources to run.
+To help setup the infrastructure resources, select what "kind" or "type" of application you want to build.`,
+		manifest.AppTypes)
+
+	if err != nil {
+		return fmt.Errorf("failed to get type selection: %w", err)
+	}
+	opts.AppType = t
+	return nil
+}
+
+func (opts *AppInitOpts) askAppName() error {
+	name, err := opts.prompt.Get(
+		fmt.Sprintf("What do you want to call this %s?", opts.AppType),
+		fmt.Sprintf(`The name will uniquely identify this application within your %s project.
+Deployed resources (such as your service, logs) will contain this app's name and be tagged with it.`, opts.projectName),
+		validateApplicationName)
+	if err != nil {
+		return fmt.Errorf("failed to get application name: %w", err)
+	}
+	opts.AppName = name
+	return nil
+}
+
+// askDockerfile prompts for the Dockerfile by looking at sub-directories with a Dockerfile.
+// If the user chooses to enter a custom path, then we prompt them for the path.
+func (opts *AppInitOpts) askDockerfile() error {
+	// TODO https://github.com/aws/amazon-ecs-cli-v2/issues/206
+	dockerfiles, err := opts.listDockerfiles()
+	if err != nil {
+		return err
+	}
+	const customPathOpt = "Enter a custom path"
+	selections := make([]string, len(dockerfiles))
+	copy(selections, dockerfiles)
+	selections = append(selections, customPathOpt)
+
+	sel, err := opts.prompt.SelectOne(
+		fmt.Sprintf("Which Dockerfile would you like to use for %s app?", opts.AppName),
+		"Dockerfile to use for building your application's container image.",
+		selections,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to select Dockerfile: %w", err)
+	}
+
+	if sel == customPathOpt {
+		sel, err = opts.prompt.Get("OK, what's the path to your Dockerfile?", "", nil)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to get Dockerfile: %w", err)
+	}
+	opts.DockerfilePath = sel
+	return nil
+}
+
+// listDockerfiles returns the list of Dockerfiles within a sub-directory below current working directory.
+// If an error occurs while reading directories, returns the error.
+func (opts *AppInitOpts) listDockerfiles() ([]string, error) {
+	wdFiles, err := afero.ReadDir(opts.fs, ".")
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	var dockerfiles []string
+	for _, wdFile := range wdFiles {
+		if !wdFile.IsDir() {
+			continue
+		}
```

Comment:
Would this ignore a `Dockerfile` in the working directory?

Reply:
yes, we suggest to run this command at the root of their workspace so I think that's fair (?)

Reply:
What if there's only a single application and the Dockerfile is in the root of the workspace?

Reply:
Hm. Folks seem to do it anyway. I think we should get rid of this restriction.

Reply:
Ok added it, but I think whenever we start working on our documentation we should have a section on our recommended way of structuring their workspace for the best UX.

```
project/
  ecs-project/
     frontend/app.yml
     backend/app.yml
  frontend/Dockerfile
  backend/Dockerfile
```